ir/src/validation/error.rs

```diff
@@ -157,6 +157,15 @@ impl SemanticError {
         ))
     }
 
+    pub(crate) fn evaluator_trace_segment_mismatch(
+        trace_segment: u8,
+        required_segment: usize,
+    ) -> Self {
+        SemanticError::InvalidUsage(format!(
+            "Evaluator function expected column in trace segment {required_segment}, but was given column in trace segment {trace_segment}.",
```

Comment:
nit: this should be `binding` instead of `column`, since it could actually be a group of multiple columns